src/games.ts

```diff
@@ -0,0 +1,273 @@
+import { ButtonInteraction, CommandInteraction, GuildMember, MessageActionRow, MessageButton, MessageEmbed, User } from 'discord.js';
+import { Colors } from './util/embed-builder';
+
+export class Buzzer {
+  interaction: CommandInteraction
+  readonly expireTime = 60000*5
+  usersBuzzed: GuildMember[] = [];
+  
+  //Buzzer
+  readonly buttonBuzzer = [
+    new MessageButton({ label: 'Buzz', customId: 'buzz', style: 'DANGER', emoji: '🐝', disabled: true }),
+  ];
+
+  //Start Pauze Reset
+  readonly buttonsBuzzerControls = [
+    new MessageButton({ label: 'Start', customId: 'start', style: 'SUCCESS'}),
+    new MessageButton({ label: 'Pause', customId: 'pause', style: 'PRIMARY'}),
+    new MessageButton({ label: 'Reset', customId: 'reset', style: 'DANGER'}),
+  ];
+
+  constructor(interaction: CommandInteraction) {
+    this.interaction = interaction;
+  }
+
+  init(): void {
+    this.interaction.reply({embeds: [this.createEmbed(Colors.warning, 'Buzzer hasn\'t started yet')],
+      components: [new MessageActionRow({components: this.buttonBuzzer}), new MessageActionRow({components: this.buttonsBuzzerControls})]});
+
+    const buzzFilter = (i: ButtonInteraction) => this.buttonBuzzer
+      .map(i => i.customId).includes(i.customId)
+      // && i.user.id == player.id
+      && i.message.interaction!.id == this.interaction.id;
+    
+    const buzzerControlFilter = (i: ButtonInteraction) => this.buttonsBuzzerControls
+      .map(i => i.customId).includes(i.customId)
+      && i.user.id == this.interaction.user.id
+      && i.message.interaction!.id == this.interaction.id;
+
+    const buzzCollector = this.interaction.channel?.createMessageComponentCollector({ filter: buzzFilter, time: this.expireTime });
+    const controlCollector = this.interaction.channel?.createMessageComponentCollector({ filter: buzzerControlFilter, time: this.expireTime });
+
+    if (!buzzCollector) return;
+    if (!controlCollector) return;
+    
+    // Collectors
+    buzzCollector.on('collect', i => {
+      buzzCollector.resetTimer();
+      controlCollector.resetTimer();
+      
+      if (this.buttonBuzzer[0].disabled) return;
```

Comment:
impossible to collect a button if the button is disabled

Reply:
I know, was extra safety check.